src/status_im/bots/events.cljs

```diff
@@ -0,0 +1,99 @@
+(ns status-im.bots.events
+  (:require [re-frame.core :refer [trim-v]]
+            [status-im.utils.handlers :refer [register-handler-db register-handler-fx]]))
+
+;;;; Helper fns
+
+(defn- chats-with-bot [chats bot]
+  (reduce (fn [acc [_ {:keys [chat-id contacts]}]]
+            (let [contacts (map :identity contacts)]
+              (if (some #{bot} contacts)
+                (conj acc chat-id)
+                acc)))
+          []
+          chats))
+
+;; TODO: optime this, for sure we don't need to re-calculate all bot subscriptions every time something in bot db changes
```

Comment:
Please add your name to the `TODO` so that we can kickly identify who will work on it: `TODO(janherich)`

Reply:
Will do